api/internal/git/repospec.go

```diff
@@ -212,18 +213,26 @@ func peelQuery(arg string) (string, string, time.Duration, bool) {
 	return parsed.Path, ref, duration, submodules
 }
 
+var userRegex = regexp.MustCompile(`^[a-zA-Z][a-zA-Z0-9-]*@`)
+
 func parseHostSpec(n string) (string, string) {
 	var host string
-	// Start accumulating the host part.
-	for _, p := range []string{
-		// Order matters here.
-		"git::", "gh:", "ssh://", "https://", "http://", "file://",
-		"git@", "github.com:", "github.com/"} {
-		if len(p) < len(n) && strings.ToLower(n[:len(p)]) == p {
-			n = n[len(p):]
-			host += p
+	consumeHostStrings := func(parts []string) {
+		for _, p := range parts {
+			if len(p) < len(n) && strings.ToLower(n[:len(p)]) == p {
+				n = n[len(p):]
+				host += p
+			}
 		}
 	}
+	// Start accumulating the host part.
+	// Order matters here.
+	consumeHostStrings([]string{"git::", "gh:", "ssh://", "https://", "http://", "file://"})
+	if p := userRegex.FindString(n); p != "" {
+		n = n[len(p):]
+		host += p
+	}
+	consumeHostStrings([]string{"github.com:", "github.com/"})
 	if host == "git@" {
```

Comment:
Playing devil's advocate with my own comment above: arguably this conditional here is our current username handling, and we could reduce special casing in the implementation by changing it to apply whenever we've found _any_ username, and allow it to consume _any_ domain, rather than just `github.com` above. That would add risk in the form of additional supported permutations we can't _specifically_ integration test... but in another way also reduce it by making use cases like the new one in this PR use the code paths we already have integration coverage for (not to mention reducing the difficulty of understanding this code). Wdyt?

cc @natasha41575 for another opinion on this tradeoff

Reply:
Acknowledging I read this comment but don't have an opinion. I'd prefer not to include that change in this PR though, since it would probably be much bigger.